### ISSUE TITLE
V5.2.0 - Dependency Bundle

### ISSUE THREAD

**Issue by heakinakai** [elastic/kibana-plugin-helpers#34]
Only the main package.json dependencies are bundled with the zip and any nested dependencies are not included in the bundle.

for example, if you `npm install --save charts.js` it also installs it's nested dependencies from it's own package.json. But the package produced by `plugin-helpers build` only contains the main dependencies, so you must go through every nested dependency of charts.js and make sure they are also listed in your main package.json.

Just for charts.js I need to include the below:
```
"dependencies": {
    "chart.js": "^2.5.0",
    "chartjs-color": "^2.1.0",
    "chartjs-color-string": "^0.4.0",
    "chartsjs-plugin-data-labels": "^0.1.5",
    "color-convert": "^1.9.0",
    "color-name": "^1.1.2",
    "color-string": "^1.5.2",
    "is-arrayish": "^0.3.1",
    "lodash.defaults": "^4.2.0",
    "moment": "^2.17.1",
    "simple-swizzle": "^0.2.2"
  }
```

I'm not sure if it's a NPM issue or something that can be fixed in the bundle process to recursively check nested dependencies .

This results in it failing the Optimizing and caching browser bundle proccess, when installing in Kibana

**ERROR in ./plugins/progress_dough/ ~ /chartjs-color/index.js
    Module not found: Error: Cannot resolve module 'chartjs-color-string' in /usr/share/kibana/plugins/progress_dough/node_modules/chartjs-color
     @ ./plugins/progress_dough/ ~ /chartjs-color/index.js 3:13-44**

**ERROR in ./plugins/progress_dough/ ~ /color-string/index.js
    Module not found: Error: Cannot resolve module 'simple-swizzle' in /usr/share/kibana/plugins/progress_dough/node_modules/color-string
     @ ./plugins/progress_dough/ ~ /color-string/index.js 3:14-39**

**Comment by heakinakai** [comment]
I've just modified the plugin_config.js on my own fork and it seems to have fixed the problem. All nested dependencies are imported.

I replaced the static join with a looping function:

I'm thinking something like this.

 // get all dependencies & nested dependencies.

  ```
function getDependencies(deps = {}){
      if (deps.length === 1) {
          buildSourcePatterns.push(`node_modules/${ deps[0] }/**/*`);
      } else if (deps.length) {
          buildSourcePatterns.push(`node_modules/{${ deps.join(',') }}/**/*`);
      }
      for(let dep of deps){
          var tmpPkg = require(resolve(root, `node_modules/${dep}/package.json`));
          var tmpDeps = Object.keys(tmpPkg.dependencies || {});
          getDependencies(tmpDeps);
      }
  }

    var deps = Object.keys(pkg.dependencies || {});

    getDependencies(deps)
```

You may want to improve this.. I'll make a pull request for you to review.

**Comment by spalger** [comment]
That works for one extra level of dependencies but we should really just update the build task to run `npm install --prod` in the build output. It will slow down the build task but it's the only way to do this correctly.

**Comment by w33ble** [comment]
This was already fixed in 6.0.0, via https://github.com/elastic/kibana-plugin-helpers/pull/28, which now runs the dependency install step on the build, instead of copying your existing dependencies.

**Comment by heakinakai** [comment]
Yep, I was just testing this, noticed the new version using yarn after trying to fix it myself... 

Always check the release tab.... :)

Guess I need to update my yeoman plugin to include the new version in the package.json by default.

Thanks @w33ble @spalger
